From 744ea767da1e8e95cf0efbcef9059b364186b1a3 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Wed, 6 Aug 2014 03:45:42 -0400
Subject: [acpi] numa: Simplify acpi_get_node() style

Message-id: <20140806034542.1438.12727.stgit@gir.stowe>
Patchwork-id: 87424
O-Subject: [RHEL7.1 PATCH 15/69] ACPI / numa: Simplify acpi_get_node() style
Bugzilla: 1110896
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110896
Upstream Status: 962fe9c91c5faa271e21c6ef589ff1df02fa13ed

commit 962fe9c91c5faa271e21c6ef589ff1df02fa13ed
Author: Bjorn Helgaas <bhelgaas@google.com>
Date:   Fri Jan 24 15:42:17 2014 -0700

    ACPI / numa: Simplify acpi_get_node() style

    Simplify control flow by removing local variable initialization and
    returning a constant as soon as possible.  No functional change.

    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>
    Acked-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/acpi/numa.c b/drivers/acpi/numa.c
index acf9ca4..98197cd 100644
--- a/drivers/acpi/numa.c
+++ b/drivers/acpi/numa.c
@@ -334,12 +334,12 @@ int acpi_get_pxm(acpi_handle h)
 
 int acpi_get_node(acpi_handle handle)
 {
- int pxm, node = NUMA_NO_NODE;
+ int pxm;
 
  pxm = acpi_get_pxm(handle);
- if (pxm >= 0 && pxm < MAX_PXM_DOMAINS)
-  node = acpi_map_pxm_to_node(pxm);
+ if (pxm < 0 || pxm >= MAX_PXM_DOMAINS)
+  return NUMA_NO_NODE;
 
- return node;
+ return acpi_map_pxm_to_node(pxm);
 }
 EXPORT_SYMBOL(acpi_get_node);
-- 
1.7.1